module/VuFind/src/VuFind/ILS/Logic/AvailabilityStatus.php

```diff
@@ -0,0 +1,287 @@
+
+    /**
+     * Check if status should be visible.
+     *
+     * @return bool
+     */
+    public function isVisible(): bool
+    {
+        return true;
+    }
+
+    /**
+     * Set use unknown message
+     *
+     * @param bool $useUnknownMessage If unknown message shall be used
+     *
+     * @return void
+     */
+    public function setUseUnknownMessage(bool $useUnknownMessage): void
+    {
+        $this->useUnknownMessage = $useUnknownMessage;
+    }
+
+    /**
+     * Check if unknown message should be used.
+     *
+     * @return bool
+     */
+    public function useUnknownMessage(): bool
+    {
+        return $this->useUnknownMessage;
+    }
+
+    /**
+     * Get status description.
+     *
+     * @return string
+     */
+    public function getStatusDescription(): string
+    {
+        if (!empty($this->status)) {
+            return $this->status;
+        }
+        switch ($this->availability) {
+            case self::STATUS_UNAVAILABLE:
+                return 'Unavailable';
+            case self::STATUS_AVAILABLE:
+                return 'Available';
+            default:
+                return 'Uncertain';
+        }
+    }
+
+    /**
+     * Get status label.
+     *
+     * @return string
+     */
+    public function getStatusLabel(): string
+    {
+        switch ($this->availability) {
+            case self::STATUS_UNAVAILABLE:
+                return 'Checked Out';
+            case self::STATUS_AVAILABLE:
+                return 'Available';
+            default:
+                return 'HoldingStatus::availability_uncertain';
+        }
+    }
+
+    /**
+     * Get status class.
+     *
+     * @return string
+     */
+    public function getStatusClass(): string
+    {
+        switch ($this->availability) {
+            case self::STATUS_UNAVAILABLE:
+                return 'danger';
+            case self::STATUS_AVAILABLE:
+                return 'success';
+            default:
+                return 'warning';
+        }
+    }
+
+    /**
+     * Get schema.
+     *
+     * @return ?string
+     */
+    public function getSchema(): ?string
+    {
+        switch ($this->availability) {
+            case self::STATUS_UNAVAILABLE:
+                return 'http://schema.org/OutOfStock';
+            case self::STATUS_AVAILABLE:
+                return 'http://schema.org/InStock';
+            default:
+                return 'http://schema.org/LimitedAvailability';
+        }
+    }
+
+    /**
+     * Get status message key
+     *
+     * @return ?string
+     */
+    public function getStatusMessageKey(): ?string
+    {
+        if ($this->useUnknownMessage) {
+            return 'unknown';
+        }
+        return lcfirst($this->getStatusDescription());
+    }
+
+    /**
+     * Get status message key.
+     *
+     * @return string
+     */
+    public function getStatusMessageTemplate(): string
+    {
+        if ($this->useUnknownMessage) {
+            return 'ajax/status-unknown.phtml';
```

Comment:
Why do you collapse most of the status templates together but leave unknown separate? Would it be better to either make one template handle all cases, or else leave the more granular setup in place?

At Villanova, we use significantly different markup for our custom unavailable template, so this refactoring would be slightly inconvenient for us (though putting an if/else in ajax/status.phtml to reimplement the customization is obviously possible). I'm just thinking that if this applies to us, it might apply to others as well, and maybe it's worth considering a more back-compatible approach.